import copy

from django_swagger_utils.drf_server.exceptions import NotFound

from ib_notifications.constants.exception_messages import INVALID_NOTIFICATION_ID, \
    INVALID_USER_NOTIFICATION_ID
from ib_notifications.exceptions.custom_exceptions import InvalidNotificationId, \
    InvalidUserNotificationId
from ib_notifications.interactors.presenters.delete_user_notification_presenter_interface import \
    DeleteUserNotificationPresenterInterface


class DeleteUserNotificationPresenter(DeleteUserNotificationPresenterInterface):

    def raise_invalid_notification_id(self, err: InvalidNotificationId):
        updated_exception = copy.deepcopy(INVALID_NOTIFICATION_ID)
        updated_exception = (
            updated_exception[0].format(err.notification_id),
            updated_exception[1]
        )
        raise NotFound(*updated_exception)

    def raise_invalid_user_notification_id(self, err: InvalidUserNotificationId):
        updated_exception = copy.deepcopy(INVALID_USER_NOTIFICATION_ID)
        updated_exception = (
            updated_exception[0].format(err.notification_id),
            updated_exception[1]
        )
        raise NotFound(*updated_exception)

    def get_success_response(self):
        return {"message": "Deleted Successfully"}
